packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Primitives/CodeWriterDeclaration.cs

```diff
@@ -2,15 +2,31 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Collections.Generic;
 
 namespace Microsoft.Generator.CSharp.Primitives
 {
     public sealed class CodeWriterDeclaration
     {
-        private string? _actualName;
-        private string? _debuggerName;
+        private Dictionary<CodeWriter.CodeScope, string> _actualNames = [];
 
-        public bool HasBeenDeclared => _actualName != null;
+        internal bool HasBeenDeclared(Stack<CodeWriter.CodeScope> scopes)
+        {
+            var top = scopes.Peek();
+            foreach (var scope in scopes)
+            {
+                if (_actualNames.ContainsKey(scope))
+                {
+                    if (!scope.Equals(top))
```

Comment:
It should resolve to the same thing, but using reference equality here seems more intuitive.